extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -544,52 +579,45 @@ class QueueProcessor extends EventEmitter {
      */
     start(options) {
         this._mProducer = new MetricsProducer(this.kafkaConfig, this.mConfig);
-        return this._mProducer.setupProducer(err => {
-            if (err) {
-                this.logger.info('error setting up metrics producer',
-                                 { error: err.message });
-                process.exit(1);
-            }
-            return this._setupProducer(err => {
-                let consumerReady = false;
+        return async.parallel([
+            done => this._mProducer.setupProducer(err => {
+                if (err) {
+                    this.logger.info('error setting up metrics producer',
+                                     { error: err.message });
+                    process.exit(1);
+                }
+                return done();
+            }),
+            done => this._setupProducer(err => {
                 if (err) {
                     this.logger.info('error setting up kafka producer',
                                      { error: err.message });
                     process.exit(1);
                 }
+                return done();
+            }),
+            done => {
                 if (options && options.disableConsumer) {
-                    this.emit('ready');
-                    return undefined;
+                    return done();
                 }
-                const groupId =
-                    `${this.repConfig.queueProcessor.groupId}-${this.site}`;
-                this._consumer = new BackbeatConsumer({
-                    kafka: { hosts: this.kafkaConfig.hosts },
-                    topic: this.repConfig.topic,
-                    groupId,
-                    concurrency: this.repConfig.queueProcessor.concurrency,
-                    queueProcessor: this.processKafkaEntry.bind(this),
-                    canary: true,
-                });
-                this._consumer.on('error', () => {
-                    if (!consumerReady) {
-                        this.logger.fatal('queue processor failed to start a ' +
-                                       'backbeat consumer');
-                        process.exit(1);
-                    }
-                });
-                this._consumer.on('ready', () => {
-                    consumerReady = true;
-                    const paused = options && options.paused;
-                    this._consumer.subscribe(paused);
-                });
-                this._consumer.on('canary', () => {
-                    this.logger.info('queue processor is ready to consume ' +
-                                     'replication entries');
-                    this.emit('ready');
-                });
-                return undefined;
-            });
+                this._consumer = this._createConsumer(
+                    this.repConfig.topic,
+                    this.processReplicationEntry.bind(this), options);
+                return this._consumer.on('canary', done);
```

Comment:
I think the canary event triggers many times per location. Maybe use `.once` here?